api/v1/stats/service.go

```diff
@@ -0,0 +1,210 @@
+					return fmt.Errorf("unmarshal to TableType error: %v", err)
+				}
+			} else {
+				return fmt.Errorf("JSON object does not have the `content` field")
+			}
+
+			stats := model.TableResult{
+				Headers: table.Headers,
+				Value:   make([]map[string]string, 0, len(table.Rows)),
+			}
+
+			for _, row := range table.Rows {
+				mp := make(map[string]string, len(row))
+				for idx, val := range row {
+					mp[table.Headers[idx]] = val
+				}
+				stats.Value = append(stats.Value, mp)
+			}
+
+			allStatsRaw = append(allStatsRaw, stats)
+			return nil
+		}()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return allStatsRaw, nil
+}
+
+func aggAppendSum(s *Service, c *gin.Context, cmd string) {
+	allStatsRaw, err := getFromCluster(s, cmd)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, adminError(err))
+		return
+	}
+
+	if len(allStatsRaw) != 0 {
+		headers := allStatsRaw[0].Headers
+
+		allStats := model.TableResult{
+			Headers: headers,
+			Value:   make([]map[string]string, 0),
+		}
+
+		for _, stats := range allStatsRaw {
+			for _, val := range stats.Value {
+				mainKey := val[headers[0]]
+
+				ok := false
+				ix := -1
+				for i, cur := range allStats.Value {
+					if cur[mainKey] == val[mainKey] {
+						ok = true
+						ix = i
+					}
+				}
+
+				if ok {
+					for k := range allStats.Value[ix] {
+						allStats.Value[ix][k] += val[k]
+					}
+				} else {
+					allStats.Value = append(allStats.Value, val)
+				}
+			}
+		}
+
+		c.JSON(http.StatusOK, allStats)
+	} else {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, adminError(fmt.Errorf("got stats of length 0")))
+		return
+	}
+}
+
+func aggSum(s *Service, c *gin.Context, cmd string) {
+	allStatsRaw, err := getFromCluster(s, cmd)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, adminError(err))
+		return
+	}
+
+	if len(allStatsRaw) != 0 || len(allStatsRaw[0].Value) != 1 {
+		headerLen := len(allStatsRaw[0].Headers)
+		acc := make([]int, headerLen)
+		for _, stats := range allStatsRaw {
+			for _, val := range stats.Value {
+				for i, k := range stats.Headers {
+					val, err := strconv.ParseInt(val[k], 10, 32)
+					if err != nil {
+						c.AbortWithStatusJSON(http.StatusInternalServerError, adminError(fmt.Errorf("%v", err)))
+						return
+					}
+					acc[i] += int(val)
+				}
+			}
+		}
+
+		headers := allStatsRaw[0].Headers
+		value := map[string]string{}
+		for i, header := range headers {
+			value[header] = fmt.Sprintf("%d", acc[i])
+		}
+		values := []map[string]string{value}
+		tab := model.TableResult{Headers: headers, Value: values}
+
+		c.JSON(http.StatusOK, tab)
+		return
+
+	} else {
+		var err errorno.ErrorResponse
+		if len(allStatsRaw) != 0 {
+			err = adminError(fmt.Errorf("got stats of length 0"))
+		} else if len(allStatsRaw[0].Value) != 1 {
+			err = adminError(fmt.Errorf("the length value should be 1"))
+		}
+		c.AbortWithStatusJSON(http.StatusInternalServerError, err)
+		return
+	}
```

Comment:
ditto, Use some short judgments to quickly short-circuit the wrong scenarios
``` go
if len(allStatsRaw) != 0 { return ...}
if len(allStatsRaw[0].Value) != 1 { return ...}

// handle normal scenario
```

Reply:
thx, [22ba412](https://github.com/hstreamdb/http-services/pull/12/commits/22ba4124c1076eaa0e7afcca97cc129ca80902ad)